crates/hyperswitch_domain_models/src/payments.rs

```diff
@@ -80,6 +80,27 @@ impl PaymentIntent {
     pub fn get_id(&self) -> &id_type::PaymentId {
         &self.payment_id
     }
+    pub fn get_order_tax_amount(&self, payment_attempt: PaymentAttempt) -> Option<MinorUnit> {
```

Comment:
This can accept a reference to the payment attempt instead?

```suggestion
    pub fn get_order_tax_amount(&self, payment_attempt: &PaymentAttempt) -> Option<MinorUnit> {
```